src/transformers/models/detr/modeling_detr.py

```diff
@@ -320,45 +322,56 @@ def replace_batch_norm(m, name=""):
         replace_batch_norm(ch, n)
 
 
-class DetrTimmConvEncoder(nn.Module):
+class DetrBackbone(nn.Module):
     """
-    Convolutional encoder (backbone) from the timm library.
+    Backbone class, using either the AutoBackbone API or one from the timm library.
 
     nn.BatchNorm2d layers are replaced by DetrFrozenBatchNorm2d as defined above.
 
     """
 
-    def __init__(self, name: str, dilation: bool, use_pretrained_backbone: bool, num_channels: int = 3):
+    def __init__(self, config):
         super().__init__()
 
-        kwargs = {}
-        if dilation:
-            kwargs["output_stride"] = 16
-
-        requires_backends(self, ["timm"])
+        self.config = config
+
+        if config.use_timm_backbone:
+            requires_backends(self, ["timm"])
+            kwargs = {}
+            if config.dilation:
+                kwargs["output_stride"] = 16
+            backbone = create_model(
+                config.backbone,
+                pretrained=config.use_pretrained_backbone,
+                features_only=True,
+                out_indices=(1, 2, 3, 4),
```

Comment:
I think we want to match the out_indices here with those in `config.out_features`

Reply:
Hmm yes, although `out_features` is highly tailored towards our Backbone API.

Can I leave it like this for now?